### ISSUE TITLE
CA1854 false positive: `TryGetValue` shouldn't be used when modifying a value type

### ISSUE THREAD

**Issue by cndouglas** [dotnet/roslyn-analyzers#6975]
### Analyzer

**Diagnostic ID**: [CA1854](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1854): `Prefer the IDictionary.TryGetValue(TKey, out TValue) method`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.401](https://dotnet.microsoft.com/download/dotnet/7.0)
```
dotnet format --version
7.4.437506+270bbfa0b86b9158575ba986dff9244ed4561192
```

### Describe the bug

The formatter incorrectly suggests using `TryGetValue()` when the code modifies a value type in the dictionary.

### Steps To Reproduce
Example code:
```C#
var dictionary = new Dictionary<string, int>();
const string key = "test";
dictionary[key] = 1;

if (dictionary.ContainsKey(key))
    dictionary[key]++;

Console.WriteLine($"value is now {dictionary[key]}");
```
Output:
```
value is now 2
```
`.editorconfig` file:
```properties
[*.cs]
dotnet_diagnostic.CA1854.severity = warning
```
1. Run `dotnet format` with the code above.

### Expected behavior

The code should not be changed.

### Actual behavior

The code is changed to the following:
```C#
var dictionary = new Dictionary<string, int>();
const string key = "test";
dictionary[key] = 1;

if (dictionary.TryGetValue(key, out int value))
    value++;

Console.WriteLine($"value is now {dictionary[key]}");
```
Output:
```
value is now 1
```

**Comment by cndouglas** [comment]
Duplicate of #6390.
